traffic_portal/test/integration/PageObjects/TopologiesPage.po.ts

```diff
@@ -0,0 +1,97 @@
+    private txtSearch = element(by.id('topologiesTable_filter')).element(by.css('label input'));
+    private txtName = element(by.name('name'));
+    private txtDescription = element(by.id('description'));
+    private btnAddCacheGroup = element(by.xpath("//a[@title='Add child cache groups to TOPOLOGY']"));
+    private txtSearchCacheGroup = element(by.id('availableCacheGroupsTable_filter')).element(by.css('label input'));
+    private btnDelete = element(by.xpath("//button[text()='Delete']"));
+    private txtConfirmName = element(by.name('confirmWithNameInput'));
+    private config = require('../config');
+    private randomize = this.config.randomize;
+    async OpenTopologiesPage(){
+        let snp = new SideNavigationPage();
+        await snp.NavigateToTopologiesPage();
+    }
+    async OpenTopologyMenu(){
+        let snp = new SideNavigationPage();
+        await snp.ClickTopologyMenu();
+    }
+    public async CreateTopologies(topologies: Topologies): Promise<boolean> {
+        let basePage = new BasePage();
+        let snp = new SideNavigationPage();
+        await snp.NavigateToTopologiesPage();
+        //click '+'
+        await this.btnCreateNewTopologies.click();
+        await this.txtName.sendKeys(topologies.Name + this.randomize)
+        await this.txtDescription.sendKeys(topologies.DescriptionData + this.randomize)
+        //click add cache group +
+        await this.btnAddCacheGroup.click();
+        //choose type
+        await this.txtCacheGroupType.sendKeys(topologies.Type);
+        await basePage.ClickSubmit();
+        //choose Cachegroup
+        await this.txtSearchCacheGroup.sendKeys(topologies.CacheGroup + this.randomize)
+        if(await browser.isElementPresent(by.xpath("//td[@data-search='^" + topologies.CacheGroup + this.randomize + "$']")) === true){
+            await element(by.xpath("//td[@data-search='^" + topologies.CacheGroup + this.randomize + "$']")).click();
+        }
+        await basePage.ClickSubmit();
+        await basePage.ClickCreate();
+        if(topologies.description === "create a Topologies with empty cachegroup (no server)"){
+            topologies.validationMessage = topologies.validationMessage + this.randomize;
+        }
+        return await basePage.GetOutputMessage().then(value => value === topologies.validationMessage);
+    }
+       
+    async SearchTopologies(nameTopologies:string): Promise<boolean>{
+        let name = nameTopologies + this.randomize;
+        let snp = new SideNavigationPage();
+        await snp.NavigateToTopologiesPage();
+        await this.txtSearch.clear();
+        await this.txtSearch.sendKeys(name);
+        if (await browser.isElementPresent(element(by.xpath("//td[@data-search='^" + name + "$']"))) == true) {
+            await element(by.xpath("//td[@data-search='^" + name + "$']")).click();
+            return true;
+        } 
+        return false;
+    }
+    async DeleteTopologies(topologies: Topologies):Promise<Boolean>{
```

Comment:
`Boolean` is the type of the constructor-used-as-a-function `globalThis.Boolean` - this will compile just fine because `boolean` is assignable to `Boolean` and the caller can check the truthiness of a `Boolean`. However, a `Boolean` is an `object`, but a `boolean` isn't. There are lots of subtle differences because of that, but mainly if you later use typeguards to check something's type like `if (typeof x === "boolean")`, you're gonna be fooled, because `typeof(false)` outputs `"boolean"` while `typeof(new Boolean())` actually outputs `"object"`.

This playground helps illustrate: https://www.typescriptlang.org/play?strict=true#code/DYUwLgBAQgXND29QEMB2EC8EwCcCuIA3ALABQokARnJYiulrgSaWQGZ6oDGYAlvOgAOOXqjABJMAAp4ceJQBWIHgEo4AN3i8AJhADeZCEYhcBAZyQgAdMHgBzGSpYBfMmWGiJ0qE-cixklKUvqykUJgQqCAA7giWaFIhlBFRsVB0IAlOQA

tl;dr use `boolean` not `Boolean` (and likewise `number` not `Number`, `string` not `String`, `object` not `Object`, `function` (or better yet strict call signature definitions) not `Function` etc.)

Reply:
For further confusion, `Boolean` when used as a function spits out `boolean`s, not `Boolean`s. For example:
```javascript
typeof(Boolean()); // "boolean"
typeof(new Boolean()); // "object"
Boolean() instanceof Boolean; // false
Boolean instanceof Boolean; // false
new Boolean() instanceof Boolean; // true;
```

Reply:
Nice catch! The package lock change I think because when I merged or npm install.